src/nodes/chunk_insert_state.c

```diff
@@ -129,19 +129,21 @@ create_compress_chunk_result_relation_info(ChunkDispatch *dispatch, Relation com
 	InitResultRelInfo(rri, compress_rel, hyper_rti, NULL, dispatch->estate->es_instrument);
 
 	rri_orig = dispatch->hypertable_result_rel_info;
-	if (rri_orig->ri_WithCheckOptions || rri_orig->ri_WithCheckOptionExprs ||
-		rri_orig->ri_junkFilter || rri_orig->ri_projectReturning)
+	/* RLS policies are not supported if compression is enabled */
+	Assert(rri_orig->ri_WithCheckOptions == NULL && rri_orig->ri_WithCheckOptionExprs == NULL);
+	if (rri_orig->ri_projectReturning)
 	{
-		elog(ERROR, "CHECK options, RETURNING clause are not supported");
+		elog(ERROR, "RETURNING clause is not supported when inserting into a compressed chunk");
```

Comment:
RETURNING is already checked ts_chunk_insert_state_create so this check seems redundant

Reply:
removed here